backend/controller/paymentController.js

```diff
@@ -182,23 +257,55 @@ const updatePlan = asyncHandler(async (req, res, next) => {
     { new: true }
   );
 
-  return res.status(200).json({ success: true, data: result });
+  return res.status(200).json({
+    statusCode: 200,
+    success: true,
+    message: "Successfully Updated the active status",
+    data: result,
+  });
 });
 
+/******************************************************
+ * @updatePlan
+ * @method DELETE
+ * @route /api/v1/payment/plan
+ * @description  delete the plan using planDocumentId
+ * @params planDocumentId
+ * @returns deleted plan object
+ ******************************************************/
+
 const deletePlan = asyncHandler(async (req, res, next) => {
   const { planDocumentId } = req.params;
 
   const result = await SubscriptionPlanModel.findByIdAndDelete(planDocumentId);
+  if (!result) return next(new customError("resource not found", 404));
 
-  // FIXME: We should check if the result is a success or not. Maybe invalid ID was provided by frontend or it does not exist in our DB
-
-  return res.status(200).json({ success: true, data: result });
+  return res.status(200).json({
+    statusCode: 200,
+    success: true,
+    message: "Successfully deleted the plan",
+    data: result,
```

Comment:
Is this sending the deleted plan details?
If yes, is it required?

Reply:
sending null in data